apps/gallery/js/ImageEditor.js

```diff
@@ -1567,35 +1574,17 @@ ImageEditor.prototype.cropImage = function(callback) {
   this.source.width = right - left;
   this.source.height = bottom - top;
 
-  this.resetPreview();
-  // Adjust the image
-  var self = this;
-  this.edit(function() {
-    // Hide and reshow the crop overlay to reset it to match the new image size
-    self.hideCropOverlay();
-    self.showCropOverlay();
+  this.requestNewPreview(function() {
     if (callback) {
       callback();
```

Comment:
If the hide and show calls aren't necessary anymore then you don't need the `if (callback)` either and can just pass the callback directly to requestNewPreview()